File: sakai-ngo/src/app/demo/service/analisis.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { analisis } from '../module/analisis';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AnalisisService {

  url : string ='http://localhost:8080/analisis'

  constructor(private HttpClient: HttpClient) { }


  crearAnalisis(Analisis: analisis, codRegistro: number):Observable<analisis>{
    const url = `${this.url}/crearAnalisis/${codRegistro}`;
     return this.HttpClient.post<analisis>(url,Analisis)
   }


   obtenerAnalisis():Observable<analisis[]>{
    return this.HttpClient.get<analisis[]>(this.url+'/listarAnalisis');
   }


   buscarXnombre(nombre: String):Observable<analisis[]>{

  return this.HttpClient.get<analisis[]>(this.url+'/buscarXNombre/'+nombre);
   }

   actualizarAnalsis(codAnalsis: number, analisis: analisis): Observable<analisis> {
    return this.HttpClient.put<analisis>(`${this.url}/modificarAnalisis/${codAnalsis}`, analisis);
  
  }

  buscaAnalisis(codAnalisis: number):Observable<analisis>{
    return this.HttpClient.get<analisis>(this.url+'/buscaAnalisis/'+codAnalisis)
  }
}

File: veterinariaf/src/main/java/com/example/veterinariaf/Service/analisisService.java
package com.example.veterinariaf.Service;

import com.example.veterinariaf.entity.analisis;
import com.example.veterinariaf.entity.analisisDTO;

import java.util.List;

public interface analisisService {


    public List<analisisDTO> listarAnalisis();

    public List<analisis> listaAnalisis();

    public analisis crearAnalisis(analisis analisa);

    public analisis modificarAnalisis(analisis analisa, int codAnalisis);

    public List<analisis> buscarAnalisisPorRegistro(int codregistro);

    public  analisis buscarAnalisis(int codAnalisis);

    public List<Object[]> buscarAnalisisXmasco(String nombre);


}
